locksmith/src/websub/jobs/locks.ts

```diff
@@ -1,19 +1,71 @@
+import { Op } from 'sequelize'
 import { Lock } from '../../graphql/datasource'
-import { Hook } from '../../models'
+import { Hook, ProcessedHookItem } from '../../models'
 import { TOPIC_LOCKS } from '../topics'
 
 import { networkMapToFnResult, notifyHook } from '../helpers'
 
+const FIRST_LIMIT = 50
+
+export async function fetchUnprocessedLocks(network: number) {
+  const lockSource = new Lock(network)
+  let page = 0
+  const items = []
+
+  while (true) {
+    const locks = await lockSource.getLocks({
+      first: FIRST_LIMIT,
+      skip: page ? page * FIRST_LIMIT : 0,
+    })
+
+    const lockIds = locks.map((lock: any) => lock.id)
+
+    const processedLocks = await ProcessedHookItem.findAll({
+      where: {
+        objectId: {
+          [Op.in]: lockIds,
+        },
+      },
+      order: [['id', 'DESC']],
+    })
+
+    const onlyUnprocessed = locks.filter(
+      (lock: any) => !processedLocks.find((item) => item.objectId === lock.id)
+    )
+
+    items.push(onlyUnprocessed)
```

Comment:
I wonder if this would not cause a pretty big memory object when we could just start processing these, but let's maybe improve later.